agent/structs/connect_proxy_config.go

```diff
@@ -256,6 +265,10 @@ type Upstream struct {
 	// an ingress gateway. This cannot and should not be set by a user, it is
 	// used internally to store the association of hosts to an upstream service.
 	IngressHosts []string `json:"-" bexpr:"-"`
+
+	// CentrallyConfigured indicates whether the upstream was defined in a proxy
+	// instance registration or whether it was generated from a config entry.
+	CentrallyConfigured bool
```

Comment:
Does this need a `json:",omitempty"` so various catalog results don't start emitting a lot of `false` values immediately?

Should you exclude this from `bexpr` filtering?

Reply:
Yes and yes